packages/fxa-auth-server/scripts/convert-customers-to-stripe-automatic-tax.ts

```diff
@@ -0,0 +1,328 @@
+    });
+
+    const report = buildReport({
+      customer,
+      subscription: firestoreSubscription,
+      product,
+      plan,
+      invoicePreview,
+    });
+
+    // TODO FXA-6581: Write report to CSV
+    console.log('report:', report);
+  } catch (e) {
+    // TODO FXA-6581: Error output formatting
+    console.error(e);
+  }
+};
+
+// Converts all eligible customers to Stripe tax given the following constraints:
+// 1. Customer must exist in the provided IP address mapping
+// 2. Stripe must be able to place the customer in a taxable location
+// 3. Upcoming subscription renewal must be more than 14 days in the future for
+//    1 month subscriptions, and more than 30 days in the future for 6 month and
+//    yearly subscriptions.
+const convertCustomersToStripeTax = async (args: {
+  isDryRun: boolean;
+  batchSize: number;
+  outputFile: string;
+}) => {
+  const config = Container.get(AppConfig);
+  const firestore = Container.get(AuthFirestore);
+  const stripe = new Stripe(config.subscriptions.stripeApiKey, {
+    apiVersion: '2022-08-01',
+    maxNetworkRetries: 3,
+  });
+
+  const collectionPrefix = `${config.authFirestore.prefix}stripe-`;
+  const subscriptionCollection = `${collectionPrefix}subscriptions`;
+
+  let startAfter: string | null = null;
+  let hasMore = true;
+
+  while (hasMore) {
+    const subscriptionSnap = await firestore
+      .collectionGroup(subscriptionCollection)
+      .where('automatic_tax.enabled', '==', false)
+      .orderBy('id')
+      .startAfter(startAfter)
+      .limit(args.batchSize)
+      .get();
+
+    startAfter = subscriptionSnap.docs.at(-1)?.data().id as string;
+    if (!startAfter) hasMore = false;
+
+    const applicableSubs = subscriptionSnap.docs
+      .map((doc) => doc.data() as FirestoreSubscription)
+      .filter(willBeRenewed)
+      .filter(isStripeTaxDisabled)
+      .filter(isWithinNoticePeriod);
+
+    for (const firestoreSubscription of applicableSubs) {
+      await generateReportForSubscription(
+        stripe,
+        firestoreSubscription,
+        args.outputFile
+      );
+    }
+  }
+};
+
+async function init() {
+  program
+    .version(pckg.version)
+    .option(
+      '-n, --dry-run [true|false]',
+      'Print what the script would do instead of performing the action.  Defaults to true.',
+      true
+    )
+    .option(
+      '-b, --batch-size [number]',
+      'Number of subscriptions to query from firestore at a time.  Defaults to 100.',
+      100
+    )
+    .option(
+      '-o, --output-file [string]',
+      'Output file to write report to. Will be output in CSV format.  Defaults to stripe-tax-existing-customers-output.csv.',
+      'stripe-tax-existing-customers-output.csv'
+    )
+    .parse(process.argv);
+
+  await setupProcessingTaskObjects('existing-customers-stripe-tax');
+
+  const isDryRun = parseDryRun(program.dryRun);
+  const batchSize = parseBatchSize(program.batchSize);
+  const outputFile = parseOutputFile(program.batchSize);
```

Comment:
`program.outputFile`? (`program.batchSize` doesn't feel right here)

Reply:
Yep - hadn't used outputFile yet so didn't catch this in testing (FXA-6581 uses this).

Reply:
That should be changed, yeah - hadn't used outputFile yet so didn't catch this in testing (FXA-6581 uses this).